src/hooks/useCredentialApi.ts

```diff
@@ -1,44 +1,170 @@
+import { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { AlertProps } from '@patternfly/react-core';
+import axios, { AxiosError, AxiosResponse, isAxiosError } from 'axios';
+import { helpers } from '../helpers';
 import { CredentialType } from '../types/types';
 
-const useCredentialApi = () => {
-  const [pendingDeleteCredential, setPendingDeleteCredential] = React.useState<CredentialType>();
-
-  /**
-   * Executes a DELETE request to delete a credential.
-   *
-   * @param {CredentialType} [credential] - (Optional) The credential to be deleted. If not provided, it uses `pendingDeleteCredential`.
-   * @returns {AxiosResponse} - The Axios response object representing the result of the request.
-   */
-  const deleteCredential = (credential?: CredentialType) => {
-    return axios.delete(
-      `${process.env.REACT_APP_CREDENTIALS_SERVICE}${(credential || pendingDeleteCredential)?.id}/`
-    );
-  };
+type ApiDeleteCredentialSuccessType = {
+  message: string;
+  deleted?: number[];
+  missing?: number[];
+  skipped?: { credential: number; sources: number[] }[];
+};
 
-  /**
-   * Deletes several selected credentials based on user interaction.
-   * Add your logic to handle the deletion of selected items within this function.
-   */
-  const onDeleteSelectedCredentials = () => {
-    const selectedItems = [];
-    console.log('Deleting selected credentials:', selectedItems);
-  };
+type ApiDeleteCredentialErrorType = {
+  detail?: string;
+  message: string;
+};
+
+const useDeleteCredentialApi = (onAddAlert: (alert: AlertProps) => void) => {
+  const { t } = useTranslation();
+
+  const apiCall = useCallback(
+    (ids: CredentialType['id'][]): Promise<AxiosResponse<ApiDeleteCredentialSuccessType>> =>
+      axios.post(`${process.env.REACT_APP_CREDENTIALS_SERVICE_BULK_DELETE}`, { ids }),
+    []
+  );
+
+  const callbackSuccess = useCallback(
+    (
+      response: AxiosResponse<ApiDeleteCredentialSuccessType>,
+      updatedCredentials: CredentialType[]
+    ) => {
+      const { data } = response;
+
+      const deletedNames = data?.deleted
+        ?.map(id => updatedCredentials.find(cred => id === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const skippedNames = data?.skipped
+        ?.map(({ credential }) => updatedCredentials.find(cred => credential === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const missingNames = data?.missing
+        ?.map(id => updatedCredentials.find(cred => id === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const missingCredsMsg = 'The following credentials could not be found:' + missingNames;
+
+      // All credentials were deleted successfully
+      if (data?.deleted?.length === updatedCredentials.length) {
+        const credentialCount = updatedCredentials.length;
+        const context = credentialCount === 1 ? 'deleted-credential' : 'deleted-credentials';
```

Comment:
required: we're working harder than we need to here, and in a few other places this looks like appears too.
i18next provides a mechanism for plurals already using `count`. let's flip here and in the other spots to `count`

> string display logic can get code verbose and confusing fast always try to shift it to i18next and the locale strings file where verbosity is expected

```
const successMessage = t('toast-notifications.description', {
   context: 'deleted-credential',
   count: updatedCredentials.length,
   name: updatedCredentials.map(({ name }) => name).join(', ')
});
```

in en.json starting on line `310` (you'll still need to confirm you don't have overlap I see another `other` already in the file on line `316`) 
```
change line 310: "description_deleted-credential": "Credential(s) were successfully deleted.",
add line: "description_deleted-credential_one": "Credential {{name}} was successfully deleted.",
change line 311: "description_deleted-credential_other": "Credentials {{name}} were successfully deleted."
...
remove and/or keep the copy if you like from line 316: "description_deleted-credential_other": "Deleted credentials {{name}} and more",
```